src/reducers/options.js

```diff
@@ -0,0 +1,21 @@
+import Immutable from 'immutable'
+
+function setOptions ({ state, type, options }) {
+  const optionsObj = {}
+  optionsObj[type] = options
+  return state.mergeIn(['options'], optionsObj)
+}
+
+import { actionReducer } from './utils'
+
+const initialState = Immutable.fromJS({ options: {} })
```

Comment:
Please provide an example (as a comment in the code) how the state is structured internally. It will help in the future ...
I even think, the 'consoleOptions' property can be listed here.

Sort of:
initialState = Immutable.fromJS({ options: { 
  consoleOptions: {} // contains vmId: { autoConnect: boolean }
} })